test/muxer.spec.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+chai.use(require('chai-checkmark'))
+const expect = chai.expect
+const sinon = require('sinon')
+
+const spdy = require('spdy-transport')
+const pair = require('pull-pair/duplex')
+const pull = require('pull-stream')
+const toStream = require('pull-stream-to-stream')
+
+const Muxer = require('../src/muxer')
+
+describe('multiplex-muxer', () => {
+  let muxer
+  let spdyMuxer
+
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  it('can be created', () => {
+    const p = pair()
+    spdyMuxer = spdy.connection.create(toStream(p), {
+      protocol: 'spdy',
+      isServer: false
+    })
+    muxer = new Muxer(p, spdyMuxer)
+  })
+
+  it('catches newStream errors', (done) => {
+    sinon.stub(spdyMuxer, 'request').callsFake((_, cb) => {
+      cb(new Error('something bad happened'))
+    })
+
+    muxer.newStream((err) => {
+      expect(err).to.exist()
+      expect(err.message).to.equal('something bad happened')
+      done()
+    })
+  })
+
+  it('catches stream errors', (done) => {
+    const stream = muxer.newStream((err) => {
+      expect(err).to.not.exist()
+      pull(
+        pull.error(Object.assign(new Error('ECONNRESET')), {
+          code: 'ECONNRESET'
+        }),
+        stream,
+        pull.onEnd((err) => {
+          expect(err).to.not.exist()
+          done()
+        })
+      )
+    })
+  })
+
+  // This logic can be removed once spdy-transport doesn't emit empty errors
```

Comment:
Should we add a `todo` here to have the linter to track this for future removal?

Reply:
Yes, good call. I will add a todo in there when I update the stream muxer dep.